### DISCUSSION TITLE
Lightspark browser plugin installing on Windows

### DISCUSSION THREAD

**Discussion by HVRproject** [lightspark/lightspark#627] [Q&A] [answered]
Hi, I'm trying to install Lightspark for running Flash content in Chrome browser. The .exe is already installed. The Lightspark web page says: "This installer will include the browser plugins, but they will not be installed automatically." What does it mean? How can I install the plugin in the Chrome browser? Thanks for answers in advance!

**Comment by dbluelle** [comment, marked as answer]
You have to put the plugin dll and the lightspark.dll in the plugin folder of the browser.
(nplightsparkplugin.dll is the plugin for Firefox, libpepflashplayer.dll is for Chrome)
For firefox the folder should be %appdata%\Mozilla\plugins.
On Chrome the plugin currently doesn't work, see #518
Be aware that the next versions of both Chrome and Firefox will remove flash support completely, so the plugins won't work anymore.

**Reply by HVRproject** [reply]
Thank you, but I can't find Mozilla\plugins folder. What is the full path?

**Reply by dbluelle** [reply]
I think you have to create the folder if it doesn't exist.
%appdata% points to the user specific application data folder.

**Reply by greentrancer** [reply]
@dbluelle I installed Lightspark-v0.8.4.1-Installer-win64.exe and copied the following files:
> libspark.dll
> libspark.dll.a
> lightspark.exe
> nplightsparkplugin.dll

to root/App/DefaultData/plugins
or root/Data/plugins

for Firefox Portable 83 (from November last year) and none ways seem to be actually running Flash content.

Do you know by any chance if I am doing something wrong?

Thank you.

![firefox_2021-03-11_00-23-52](https://user-images.githubusercontent.com/5639078/110717043-40943e00-8200-11eb-8d9a-2c7a91050f9c.png)

**Reply by dbluelle** [reply]
Hmm, according to your screenshot firefox seems to have detected the plugin, no idea why it doesn't work.
What happens on the page you are testing?

**Reply by KrasnayaPloshchad** [reply]
Me too on Pale Moon 29.1.0.

**Reply by swatwik-dev** [reply]
> You have to put the plugin dll and the lightspark.dll in the plugin folder of the browser.
> (nplightsparkplugin.dll is the plugin for Firefox, libpepflashplayer.dll is for Chrome)
> For firefox the folder should be %appdata%\Mozilla\plugins.
> On Chrome the plugin currently doesn't work, see #518
> Be aware that the next versions of both Chrome and Firefox will remove flash support completely, so the plugins won't work anymore.

I have installed the Lightspark-v0.8.4.1-Installer-win64.exe in my windows pc. After that to install the Lightspark I have created the directory 'appdata\Mozilla\plugins' in 'C:\Program Files\Mozilla Firefox\'. After I have copied the following files.

libspark.dll
libspark.dll.a
lightspark.exe
nplightsparkplugin.dll

Now how can I see that the Lightspark plugin has been installed in the firefox. Please let me know the way for this.

Please help me.

**Comment by greentrancer** [comment]
For example I am trying to play this game: http://www.nitrome.com/games/flashcat/

All that I get for this it is a white screen and nothing happens.

**Reply by dbluelle** [reply]
I've tried that on waterfox, I also get a white screen but after a few minutes(!) the game is displayed.
I have to examine why that takes so long.

**Reply by greentrancer** [reply]
Thank you for looking into it, much appreciated!

**Reply by dbluelle** [reply]
I've reduced the loading time with the latest commits.
The swf tries to establish a connection to a server that doesn't properly respond. Apparently Adobe has a timeout of 3 seconds in this case, so we do the same.

**Reply by greentrancer** [reply]
Thank you very much for the update, wish you all the best!

**Comment by BlueMints000** [comment]
Hi, will LightSpark work in the background for alternative browsers? Or would I have to put LightSpark on that browser.

**Reply by dbluelle** [reply]
You have to install the lightspark plugin in the plugin folder of the alternative browser (which may or may not be the same as on Firefox/Chrome)

**Reply by BlueMints000** [reply]
ok cool, Thank you

**Comment by Purplstuph** [comment]
How do i install this on opera since this doesn't work with chrome?

**Reply by dbluelle** [reply]
AFAIK opera uses the chrome web engine, so unless opera does something weird, following the instructions for chrome should probably work.

**Comment by huwdp** [comment]
If anyone has gotten Lightspark to work under Chrome or Firefox then please could you add instructions here (https://github.com/lightspark/lightspark/wiki/Getting-Lightspark-up-and-running-in-Web-Browser )
